packages/runtime-core/src/vnode.ts

```diff
@@ -593,8 +593,9 @@ function _createVNode(
 export function guardReactiveProps(props: (Data & VNodeProps) | null) {
   if (!props) return null
   return isProxy(props) || InternalObjectKey in props
-    ? extend({}, props)
+    ? JSON.parse(JSON.stringify(props)) as (Data & VNodeProps)
```

Comment:
that's not a good idea. props could contain `Map` or `Set` objects as well as user-provided class instances. Those are not serializable.

Reply:
> that's not a good idea. props could contain `Map` or `Set` objects as well as user-provided class instances. Those are not serializable.
You are right. I am surprised to find that the result of using 'mergeProps' is normal, but the new object returned by' mergeprops' is not a reactive object. I feel this is also not a good method. But if you add another attribute to div (`<div v-bind= "config" attrs>< /div>`), the compilation result will also call 'mergeProps'. Maybe we can solve this problem from the compiler